routes/referenda.js

```diff
@@ -0,0 +1,190 @@
+  res.render('referenda/index.pug', { req, activeReferenda, ballotsCast, referendaCompleted })
+})
+
+app.get(__l('/referenda/active'), function (req, res) {
+  const data = loadActiveReferenda()
+  res.render('referenda/active.pug', { req, referenda: data })
+})
+
+app.get(__l('/referenda/complete'), function (req, res) {
+  const data = loadCompleteReferenda()
+  res.render('referenda/complete.pug', { req, referenda: data })
+})
+
+app.get(__l('/referendum/create'), function (req, res) {
+  res.render('referenda/create.pug', { req })
+})
+
+app.post(__l('/referendum/create'), function (req, res) {
+  const referendum = {
+    title: req.body.title,
+    slug: slug(req.body.title),
+    deadline: req.body.deadline,
+    binding: req.body.binding === 'on',
+    amendable: req.body.amendable === 'on',
+    published: false,
+    createdBy: req.user
+  }
+
+  if (!referendum.title || !referendum.deadline) {
+    res.status(400).render('error.pug', {req, msg: __('referendum-create-fail')})
+  } else {
+    const data = referenda.load()
+    const existing = data.find(ref => ref.slug === referendum.slug)
+    if (existing !== undefined) {
+      res.status(400).render('error.pug', {req, msg: __('referendum-create-fail')})
+    }
+
+    data.push(referendum)
+    referenda.save(data)
+    res.redirect(`/referendum/${referendum.slug}/edit`)
+  }
+})
+
+app.get(__l('/referendum/:slug'), function (req, res) {
+  const data = referenda.load()
+  const referendum = data.find(ref => ref.slug === req.params.slug)
+  if (referendum === undefined) {
+    return res.status(404).render('error.pug', {req, err: 'No such referendum'})
+  }
+  const participants = referendum.participants || []
+  const isActive = isReferendumActive(referendum)
+  const hasVoted = participants.includes(req.user)
+  const canVote = isActive && !hasVoted
+  const votes = referendum.votes || {}
+
+  const candidates = referendum.candidates || []
+  const resultsVisible = true
+
+  res.render('referenda/referendum.pug', { req, referendum, isActive, canVote, candidates, resultsVisible, votes })
+})
+
+app.post(__l('/referendum/:slug'), function (req, res) {
+  const data = referenda.load()
+  const referendum = data.find(ref => ref.slug === req.params.slug)
+  if (referendum === undefined) {
+    return res.status(404).render('error.pug', {req, err: 'No such referendum'})
+  }
+  const participants = referendum.participants || []
+  const isActive = isReferendumActive(referendum)
+  const hasVoted = participants.includes(req.user)
+  const canVote = isActive && !hasVoted
+  if (!canVote) {
+    return res.status(403).render('error.pug', {req, err: 'Cannot vote in this referendum'})
+  }
+  const candidates = referendum.candidates || []
+
+  const votes = candidates.reduce((votes, candidate) => {
+    const userVote = req.body[candidate] === 'on' ? 1 : 0
+    votes[candidate] = (votes.hasOwnProperty(candidate) ? votes[candidate] : 0) + userVote
```

Comment:
Newer eslint [deprecates using `.hasOwnProperty` directly](https://eslint.org/docs/rules/no-prototype-builtins).
```suggestion
    votes[candidate] = Object.prototype.hasOwnProperty.call(votes, candidate) ? votes[candidate] : 0) + userVote
```